### PR TITLE
Refactoring Task.java, restructuring DataLoader.java and updating navigation bar looks

### PR DESCRIPTION
**Pull request by Decitizen**

### CODE REVIEW

**Comment by dwarfcrank** [comment]
This PR seems to be made against the master branch, judging from the first two commits, dunno if they should end up in the dev branch.

File: src/main/java/pingis/services/DataLoader.java [hunk 1]
```diff
@@ -11,117 +13,111 @@
 
 @Component
 public class DataLoader implements ApplicationRunner {
-//CHECKSTYLE:OFF
     private ChallengeRepository cr;
     private TaskRepository tr;
+    private LinkedHashMap<Challenge, ArrayList<Task>> challenges;
```
Comment (dwarfcrank) [inline src/main/java/pingis/services/DataLoader.java:18]: Why `LinkedHashMap` specifically?

File: src/main/resources/templates/navbar.html [hunk 2]
```diff
@@ -8,46 +8,46 @@
         <span class="icon-bar"></span>
         <span class="icon-bar"></span>
       </button>
-      <a class="navbar-brand" href="#">TDD-Pingis</a>
+        <a class="navbar-brand" href="#">TDD-pingpong</a>
     </div>
-
+
```
Comment (dwarfcrank) [inline src/main/resources/templates/navbar.html:13]: Try to avoid introducing unnecessary whitespace changes. `git diff --check` is useful for checking these.

File: src/test/java/pingis/entities/TaskTest.java [hunk 3]
```diff
@@ -11,15 +11,19 @@
...
+                1, 0);
         protectedTask = new Task();
     }
 
     @Test
     public void testTaskCode() {
         assertEquals("generate toString()", normalTask.getDesc());
     }
-
+
```
Comment (dwarfcrank) [inline src/test/java/pingis/entities/TaskTest.java:26]: Same here re: whitespace.

File: src/main/java/pingis/services/DataLoader.java (hunk 1 above)
Comment (Decitizen) [reply src/main/java/pingis/services/DataLoader.java:18]: This makes it possible for us to loop through the HashMap structure in predictable ordering. 
* In regular ```HashMap``` the keys are not ordered in terms of their addition order. 
* By using ```LinkedHashMap``` we can loop and access the keys in the same order we put them into the data structure. This is helpful for the Challenges' id's.

File: src/main/resources/templates/navbar.html (hunk 2 above)
Comment (Decitizen) [reply src/main/resources/templates/navbar.html:13]: Aye, noted.
